src/backend/distributed/metadata/pg_get_object_address_12_13_14.c

```diff
@@ -0,0 +1,523 @@
+/*-------------------------------------------------------------------------
+ *
+ * pg_get_object_address_12_13_14.c
+ *
+ * Copied functions from Postgres pg_get_object_address with acl/owner check.
+ * Since we need to use intermediate data types Relation and Node from
+ * the pg_get_object_address, we've copied that function from PG code and
+ * added required owner/acl checks for our own purposes.
+ *
+ * We need to make sure that function works with future PG versions. Update
+ * the function name according to supported PG versions as well.
+ *
+ * Copyright (c) Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "postgres.h"
+#include "miscadmin.h"
+
+#include "catalog/objectaddress.h"
+#include "catalog/pg_type.h"
+#include "distributed/citus_ruleutils.h"
+#include "distributed/metadata/dependency.h"
+#include "distributed/metadata/distobject.h"
+#include "distributed/pg_version_constants.h"
+#include "utils/array.h"
+#include "utils/builtins.h"
+#include "parser/parse_type.h"
+
+static void ErrorIfCurrentUserCanNotDistributeObject(ObjectType type,
+													 ObjectAddress *addr,
+													 Node *node,
+													 Relation *relation);
+static List * textarray_to_strvaluelist(ArrayType *arr);
+
+/* It is defined on PG >= 13 versions by default */
+#if PG_VERSION_NUM < PG_VERSION_13
+	#define TYPALIGN_INT 'i'
+#endif
+
+/*
+ * PgGetObjectAddress gets the object address. This function is mostly copied from
+ * pg_get_object_address of the PG code. We need to copy that function to use
+ * intermediate data types Relation and Node to check acl or ownership.
+ *
+ * Codes added by Citus are tagged with CITUS CODE BEGIN/END.
+ */
+ObjectAddress
+PgGetObjectAddress(char *ttype, ArrayType *namearr, ArrayType *argsarr)
+{
```

Comment:
We should roughly have a test file like this: https://gist.github.com/onderkalaci/9689f46727f62220a9761315d454db1d